src/classes/RD2_ValidationService_TEST.cls

```diff
@@ -200,6 +212,208 @@ private with sharing class RD2_ValidationService_TEST {
+
+        Contact con = getContact();
+
+        TEST_RecurringDonationBuilder rdBuilder = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withDefaultValues()
+            .withDayOfMonth('15')
+            .withStatusClosed();
+
+        for (Decimal amount : new Set<Decimal>{
+            null,
+            0,
+            -10
+        }) {
+            npe03__Recurring_Donation__c rd = rdBuilder
+                .withAmount(amount)
+                .build();
+
+            String errMessage = '';
+            try {
+                insert rd;
+            } catch (Exception e) {
+                errMessage = e.getMessage();
+            }
+            System.assert(String.isBlank(errMessage), 'Recurring Donation should be created successfully: ' + errMessage);
+        }
+    }
+
+    /***
+    * @description Verifies if it displays error message when installment period
+        is monthly and day of month is blank
+    */
+    @isTest
+    private static void shouldFailWhenDayOfMonthIsBlankForMonthlyInstallmentPeriod() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withAmount(10)
+            .withInstallmentPeriodMonthly()
+            .build();
+
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+
+        System.assert(errMessage.contains(System.Label.RD2_DayOfMonthIsRequiredForMonthlyInstallment));
+
+    }
+
+    /***
+    * @description Verifies if it displays error message when installment period
+        is monthly and day of month is blank
+    */
+    @isTest
+    private static void shouldNotCreateRDWhenInstallmentPeriodIsNotValid() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withAmount(10)
+            .withInstallmentPeriod('Occasionally')
+            .build();
+
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+        String expectedMessage = String.format(
+            System.Label.RD2_InstallmentPeriodMustBeValid,
+            new String[]{'Occasionally' }
+        );
+        System.assert(errMessage.contains(expectedMessage));
+
+    }
+
+    /***
+    * @description Verifies if it displays error message when installment period
+        is monthly and day of month value is not valid
+    */
+    @isTest
+    private static void shouldFailWhenDayOfMonthIsNotValidForMonthlyInstallments() {
```

Comment:
Can we validate more invalid "Day of Month" values similarly what we did for Amount validation in `shouldNotCreateActiveRDWhenAmountIsNotValid()`? For example, invalid values can be : 29, 30, 31, 0, -1, 'foo'.

Reply:
@RadSFDeveloper , Having invalid values like 0,-1 or 'foo' invoke invalid picklist value error because it is a restricted picklist. I am going to check for values till 31.

Reply:
Sounds good, thank you @meena-sfdc !